src/Microsoft.DotNet.Cli.Utils/BuiltInCommand.cs

```diff
@@ -52,6 +54,11 @@ public CommandResult Execute()
                     // Reset the Reporters to the new Console Out and Error.
                     Reporter.Reset();
 
+                    if (!string.IsNullOrEmpty(_workingDirectory))
+                    {
+                        Directory.SetCurrentDirectory(_workingDirectory);
```

Comment:
I see my mistake.  You were just saying "like `temp-out`".  "It could be a new argument like this existing one."

I can go either way on this one.  But a part of me kind of thinks this is the right fix.  We are invoking `dotnet-compile-{language}` and instead of passing in a `this-is-your-working-directory` argument, we actually set the working directory. It's kind of what WorkingDirectory is there for.

Thoughts?